pkg/controller/runGC_test.go

```diff
@@ -0,0 +1,256 @@
+package controller
+
```

Comment:
I don't think I see a test where we're making sure the GC doesn't touch non-failed/succeeded builds (i.e. those running or pending)